### ISSUE TITLE
[typescript] d.ts definition returning object splits lines early

### ISSUE THREAD

**Issue by dsifford** [prettier/prettier#6215]
**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACAMwFcowYBLaQiCACnjRiTwGcZtyoBzASmeAA6UPHgBiNZmw7cA3EJEAhAIbZJ7TlzlQAvjJAAaEBAwVoLZKBXYIAdwAKKhOZRKAbhHIATAyABG2JTAAazgYAGUMQI1kdiI4QwALGABbABsAdQTyeBZIsDgwp2zyV2yAT2RwFnNDThY4XDsArmSlZAIlVPrDACsWNAUA4NCwpWS4ABlOOHbO7pA+tDCNVLgARSIIeFmu+JBI7HrsSpgyjDgWMA4THwxpGHSvGATkAA4ABkM7iHr0gIxKncLg1XDNDABHTbwJrGZwgJQsAC0UDgcE8aJ82DgkPIWKaShabSQHV2hnqyXIMWwcTJK3WUJmxLmexgSl8j08z2QACZDOwlORUhoAMIQZKtSpQaBgkBEeoAFTZzhJ9W02iAA)
```sh
--parser typescript
```

**Input:**
```tsx
export function foo(text: string): {
  Foo: string;
  Bar: string;
};
```

**Output:**
```tsx
export function foo(
  text: string
): {
  Foo: string;
  Bar: string;
};

```

**Expected behavior:**

I would expect the formatted output to look like below, as the line width is well below 80 characters.

```tsx
export function foo(text: string): {
  Foo: string;
  Bar: string;
};
```

**Comment by brainkim** [comment]
I could take a look at this one over the weekend if no one gets to it. My questions are:
Do people actually run prettier over d.ts files?

Related issues and PRs:
#2581
#4070
#4228
#6099 
#6106

**Comment by dsifford** [comment]
@brainkim 

> Do people actually run prettier over d.ts files?

Yes --> https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/.prettierrc.json

> Is it actually preferable to format the function like that? Doesn’t this make the return type look like a function body?

What does the alternative look like?

**Comment by brainkim** [comment]
@dsifford
> What does the alternative look like?

The alternative would be the current behavior. If you imagine foo had a function body:
```ts
export function foo(
  text: string
): {
  Foo: string;
  Bar: string;
} {
  return bar();
}
```
prettier would change fewer lines, whereas with the suggested behavior, prettier would change representation between functions with and without a function body.

**Comment by dsifford** [comment]
Got it. Well, for me personally, I still prefer the format I suggested as it's a bit easier (for me) to grok quickly. But majority rules! :+1:

**Comment by brainkim** [comment]
I agree. One thing to note is that your example doesn’t actually extend beyond 80 characters and is split because of the (maybe overly aggressive) heuristic introduced in #4228. And even if it extended past the line limit I think the function would still be prettier as:

```js
export function foo(
  text: string
): { Foo: string; Bar: string; } {
  return bar();
}
```
where the return type is kept on a single line because it fits on a single line. Dunno.
